packages/react-instantsearch-dom-maps/src/Provider.js

```diff
@@ -19,38 +18,28 @@ class Provider extends Component {
     currentRefinement: BoundingBoxPropType,
   };
 
-  static childContextTypes = {
-    [STATE_CONTEXT]: PropTypes.shape({
-      isRefineOnMapMove: PropTypes.bool.isRequired,
-      hasMapMoveSinceLastRefine: PropTypes.bool.isRequired,
-      toggleRefineOnMapMove: PropTypes.func.isRequired,
-      setMapMoveSinceLastRefine: PropTypes.func.isRequired,
-      refineWithInstance: PropTypes.func.isRequired,
-    }).isRequired,
-  };
-
   isPendingRefine = false;
   boundingBox = null;
   previousBoundingBox = null;
 
-  getChildContext() {
-    const {
-      isRefineOnMapMove,
-      hasMapMoveSinceLastRefine,
-      toggleRefineOnMapMove,
-      setMapMoveSinceLastRefine,
-    } = this.props;
+  refineWithInstance = instance => {
+    const { refine } = this.props;
 
-    return {
-      [STATE_CONTEXT]: {
-        refineWithInstance: this.refineWithInstance,
-        toggleRefineOnMapMove,
-        setMapMoveSinceLastRefine,
-        isRefineOnMapMove,
-        hasMapMoveSinceLastRefine,
-      },
-    };
-  }
+    const bounds = instance.getBounds();
+
+    refine({
+      northEast: bounds.getNorthEast().toJSON(),
+      southWest: bounds.getSouthWest().toJSON(),
+    });
+  };
+
+  mapValue = {
```

Comment:
I said something that was wrong on the previous review. The value `isRefineOnMapMove` and `hasMapMoveSinceLastRefine` do change across render. The rest is not though we can keep the same reference. We have to update their values otherwise since we always re-use the same object the `Consumer`s don't re-render. We do have the issue on the stories: `Control` and `Redo` don't work anymore.